### DISCUSSION TITLE
Strict/-Werror mode

### DISCUSSION THREAD

**Discussion by MarkusTeufelberger** [ansible/ansible-lint#1742] [Ideas]
There are a lot of options in ansible-lint that allow exceptions from rules, but for example running it in `pre-commit` anything that is not an error will not even be displayed to the user and just silently be ignored.

It would be helpful to be able to make the linter return a nonzero return code on any warning too, just like on errors.

**Comment by MarkusTeufelberger** [comment]
Also log messages are inconsistent as of 6.7.0:

Example:
`WARNING  Listing 61 violation(s) that are fatal`
[list of warnings and errors, both looking exactly the same with no distinction which is which]
`Finished with 50 failure(s), 11 warning(s) on xxx files.`

With "fatal violations", I'd expect that the linter would actually return a nonzero code. If these were 61 warnings though, it would just return 0 and `pre-commit` would report success. It is also not possible to distinguish warnings and errors ("failures"?) in the output.

**Reply by ssbarnea** [reply]
I will add this feature for you, just vote on https://github.com/ansible/ansible-lint/discussions/2530 about which arguments we should pick.

**Reply by ssbarnea** [reply]
https://github.com/ansible/ansible-lint/pull/2533 should fix it. If you can test it, even better.
